core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -444,6 +444,12 @@ private void validateNoNewDeletesForDataFiles(
     if (parent == null || base.formatVersion() < 2) {
       return;
     }
+    List<Snapshot> snapshots =
+        Lists.newArrayList(
+            SnapshotUtil.ancestorsBetween(
+                base.currentSnapshot().snapshotId(), startingSnapshotId, base::snapshot));
+    boolean ignorePositionDeletes =
+        snapshots.stream().allMatch(this::cannotContainPosDeletesForPreviousSnapshots);
```

Comment:
Another issue came to my mind is that when there are overwrite operation snapshots committed by other engines, this will return `false` and the next verification step will still validate the pos deletes committed by Flink.

Reply:
To avoid this, we can read delete manifests only from snapshots that do not have "position-deletes-within-commit-only"="true". I can work on this change if the PR is to be merged.